src/GitHub.App/Controllers/NavigationController.cs

```diff
@@ -66,7 +67,15 @@ public void LoadView(IConnection connection, ViewWithData data, Action<IView> on
             switch (data.MainFlow)
             {
                 case UIControllerFlow.PullRequestCreation:
+                    //The selected flow will change after the view is created, so save it.
+                    var previousFlow = Current?.SelectedFlow;
+
                     CreateView(connection, data, onViewLoad);
+                    if (data.Data == null && previousFlow == UIControllerFlow.PullRequestCreation && history.Count >= 2)
+                    {
+                        Pop(history[history.Count-2]);
```

Comment:
Instead of popping after the fact, you should pop the current view before calling `CreateView` if the current and the new flow are the same. That will ensure you don't get duplicates without having this harcoded value here.